include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl

```diff
@@ -816,6 +841,27 @@ struct smoothStep_helper<T NBL_PARTIAL_REQ_BOT(VECTOR_SPECIALIZATION_CONCEPT) >
 	}
 };
 
+template<typename T, typename U>
+NBL_PARTIAL_REQ_TOP(VECTOR_SPECIALIZATION_CONCEPT && vector_traits<T>::Dimension == vector_traits<U>::Dimension)
+struct mix_helper<T, U NBL_PARTIAL_REQ_BOT(VECTOR_SPECIALIZATION_CONCEPT && vector_traits<T>::Dimension == vector_traits<U>::Dimension) >
+{
```

Comment:
remind me, `VECTOR_SPECIALIZATION_CONCEPT` is all vectors in C++ and only non-native vectors in HLSL correct?

then you have a coverage hole of mix of vector T and boolean vector U in HLSL

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/blob/new_tgmath/include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl#L215

my previous partial specialization requires all function parameters to be of the same type

Reply:
done